src/twitter/twitter_data_mappings.bal

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function convertToStatus(json jsonStatus) returns Status {
+    Status status = {};
+    var createdAt = jsonStatus.created_at;
+    if (createdAt is json) {
+        status.createdAt = createdAt != null ? createdAt.toString() : "";
+    }
+    var id = jsonStatus.id;
+    if (id is json) {
+        status.id = id != null ? convertToInt(id) : 0;
+    }
+    var text = jsonStatus.text;
+    if (text is json) {
+        status.text = text != null ? text.toString() : "";
+    }
+    var sourceOut = jsonStatus.'source;
+    if (sourceOut is json) {
+        status.'source = sourceOut != null ? sourceOut.toString() : "";
+    }
+    var truncated = jsonStatus.truncated;
+    if (truncated is json) {
+        status.truncated = truncated != null ? convertToBoolean(truncated) : false;
+    }
+    var inReplyToStatusId = jsonStatus.in_reply_to_status_id;
+    if (inReplyToStatusId is json) {
+        status.inReplyToStatusId = inReplyToStatusId != null ? convertToInt(inReplyToStatusId) : 0;
+    }
+    var favorited = jsonStatus.favorited;
+    if (favorited is json) {
+        status.favorited = favorited != null ? convertToBoolean(favorited) : false;
+    }
+    var retweeted = jsonStatus.retweeted;
+    if (retweeted is json) {
+        status.retweeted = retweeted != null ? convertToBoolean(retweeted) : false;
+    }
+    var favouritesCount = jsonStatus.favourites_count;
+    if (favouritesCount is json) {
+        status.favouritesCount = favouritesCount != null ? convertToInt(favouritesCount) : 0;
+    }
+    var retweetCount = jsonStatus.retweet_count;
+    if (retweetCount is json) {
+        status.retweetCount = retweetCount != null ? convertToInt(retweetCount) : 0;
+    }
+    var lang = jsonStatus.lang;
+    if (lang is json) {
+        status.lang = lang != null ? lang.toString() : "";
+    }
+    var geo = jsonStatus.geo;
+    if (geo is json) {
+        status.geo = geo != null ? convertToGeoLocation(geo) : {};
+    }
+    return status;
+}
+
+function convertToInt(json jsonVal) returns int {
+    var intVal = typedesc<int>.constructFrom(jsonVal);
```

Comment:
What do we expect as `jsonVal` here? Numeric values or strings also?

`constructFrom()` doesn't work for string to int.

For numeric values it would be better to do something like
```ballerina
import ballerina/lang.'int;

function convertToInt(json jsonVal) returns int {
    if (jsonVal is int) {
        return jsonVal;
    } else if (jsonVal is float|decimal) {
        return <int> jsonVal; // may have to trap for overflow
    } else if (jsonVal is string) { // can remove this if invalid
        int|error value = 'int:fromString(jsonVal);
        if (value is int) {
            return value;
        }
    }
    panic error("invalid value");
}
```

Reply:
Please check others too, it's better to narrow down to the most accurate cases.